File: src/main/java/com/efedotov/vaultly/dto/folder/FolderCreateDto.java
package com.efedotov.vaultly.dto.folder;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;

import java.util.UUID;

@Data
public class FolderCreateDto {
    @NotBlank(message = "Имя папки обязательно")
    private String name;

    private UUID parentFolderId;

    private Boolean isHidden = false;

    private String hiddenFolderKey;

    private Boolean isPrivate = true;

    private String password;

    private String description;
}

File: lib/server/model/folder/folder_create_dto/folder_create_dto.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'folder_create_dto.freezed.dart';
part 'folder_create_dto.g.dart';

@freezed
abstract class FolderCreateDto with _$FolderCreateDto {
  const factory FolderCreateDto({
    required String name,
    String? parentFolderId,
    bool? isHidden,
    String? hiddenFolderKey,
    bool? isPrivate,
    String? password,
    String? description,
  }) = _FolderCreateDto;

  factory FolderCreateDto.fromJson(Map<String, dynamic> json) =>
      _$FolderCreateDtoFromJson(json);
}